authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHDFSPlugin.java

```diff
@@ -204,48 +204,35 @@ protected void updatePolicyByMetadataObject(
       AuthorizationMetadataObject newAuthzMetaObject) {
     PathBasedMetadataObject newPathBasedMetadataObject =
         (PathBasedMetadataObject) newAuthzMetaObject;
-    List<RangerPolicy> oldPolicies = wildcardSearchPolies(oldAuthzMetaObject);
-    List<RangerPolicy> existNewPolicies = wildcardSearchPolies(newAuthzMetaObject);
-    if (oldPolicies.isEmpty()) {
+    RangerPolicy oldPolicy = findManagedPolicy(oldAuthzMetaObject);
+    RangerPolicy existNewPolicy = findManagedPolicy(newAuthzMetaObject);
+    if (oldPolicy == null) {
```

Comment:
I not sure.
The `wildcardSearchPolies()` will use HDFS path prefix to search Ranger Policy, maybe return some matching Ranger Polices. For example, When I use `/a` to search, will return `/a/b1`, `/a/b2` and `/a/b/c`?
Please notice and verify this case.

Reply:
Yes, wildcardSearchPolies will return multiple results. So I used precise search now. I think it's ok now.